### PR TITLE
feat(ext/net): Deno.upgradeHttp handles unix connections

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: cli/tests/unit/http_test.ts [hunk 1]
```diff
@@ -1782,6 +1782,120 @@ Deno.test("upgradeHttp", async () => {
...
+        const secondPacketText = new TextDecoder().decode(buf.slice(0, n));
+        assertEquals(secondPacketText, "bla bla bla\nbla bla\nbla\n");
+        conn.close();
+      })();
+
+      const resp = new Response(null, { status: 101 });
+      await respondWith(resp);
+      await promise;
+      httpConn.close();
+      break;
```
Comment (ry) [inline cli/tests/unit/http_test.ts:1892]: Seems weird to have a for await loop that breaks after one iteration.

There's a lot of sloppy loops in the tests that need to be unrolled and simplified - I expect the line-of-code could be halved.

**Review by ry — APPROVED** [review]
LGTM

File: cli/tests/unit/http_test.ts (hunk 1 above)
Comment (bartlomieju) [reply cli/tests/unit/http_test.ts:1892]: Opened https://github.com/denoland/deno/issues/13988

**Comment by KnorpelSenf** [comment]
I believe this PR to be causing #14437. Is that possible?

**Comment by bartlomieju** [comment]
> I believe this PR to be causing #14437. Is that possible?

This change is unrelated to the `fetch` API.

**Pull request merged**